### ISSUE TITLE
Using StreamProvider with BehaviourSubject from DartRx

### ISSUE THREAD

**Issue by matejthetree** [rrousselGit/provider#417]
I am using Provider library for dependency injection.

In my BusinessLogic I have BehaviourSubjects from DartRx where I hold Data.

example: 

```
class ExampleModel {
  final BehaviourSubject<ExampleObservable> example$ = BehaviourSubject();
}
```

First I map that model 

```
Provider<ExampleModel>(
  create: (context) => ExampleModel()
)
```

and then I map that stream

```
StreamProvider<ExampleObservable>(
  create: (context) => Provider.of<ExampleModel>(context, listen: false).example$
)
```

It works great as I just provide observables that can be consumed in the app, and I don't worry about rebuilding.

Now the one thing I get is that first time builder gets called in consumer, the value is always null.

Behaviour subject though keeps the last emited value, and this could be circumvented.

I was thinking of 2 ways this could be done.

1. Create a BehaviourSubjectProvider that will check for the value of the subject, to see if there is data saved. 
2. Somehow get a context in initialData, that way we can provide inital data from the provider, and keep using the StreamProvider

I am open to suggestions, and I am ready to implement this myself. Any help would be aprecciated

**Comment by awhitford** [comment]
I too am running into an issue with `FutureProvider` where the first request gets a `null` (`initialData`) where I would prefer to just wait on the `create`.  I erroneously thought that if `initialData` was not specified, then it wasn't being used, but alas it is anyway.  I was thinking of a flag to say whether to use `initialData`.

**Comment by rrousselGit** [comment]
The problem is, what is Provider.of supposed to return if the value hasn't loaded yet?

**Comment by bleroux** [comment]
@matejthetree 
It looks like your ExampleModel class should provide an initial value.
It can be done using 'seeded' constructor :
```dart
 final subject = BehaviorSubject<int>.seeded(1);
```

**Comment by matejthetree** [comment]
Doesn't work. I still get null. Even if I seed the value.

**Comment by matejthetree** [comment]
If there is nothing in behaviour subject, then null. But if it holds a
value, then that value should act as initial value.

**Comment by rrousselGit** [comment]
Closing this as there is no obvious action to do

Feel free to make a new issue with more detail about what you want if there's still a problem
